src/sentry/services/hybrid_cloud/organization/service.py

```diff
@@ -189,6 +189,24 @@ def get_organization_by_slug(
             return None
         return org_context
 
+    def get_organization_by_slug_or_id(
+        self, *, identifier: str, only_visible: bool, user_id: Optional[int] = None
+    ) -> Optional[RpcUserOrganizationContext]:
+        from sentry.models import OrganizationStatus
+
+        org_id = self.check_organization_by_slug(slug=identifier, only_visible=only_visible)
```

Comment:
We don't prevent overlaps between slugs and ids. It is possible for an organization to shadow the id of a different organization with their slug.  I'm with @RyanSkonnord in that we should have separate endpoints or parameters for id & slug so that we don't have shadowing issues.

Reply:
I agree for now to separate them but not for long term. I think it's a better long term approach if we don't let orgs have numerical slugs at all and move all (at least important to partner) endpoints to taking both id and slug. 

For this specific change I split to two because it seems like they are a bunch 🤦‍♀️